### PR TITLE
Move template methods into files and require them

### PR DESCRIPTION
**Pull request by Ru-T**
## Why?
- To get logic out of template.rb file
## What Changed?
- Moved all methods into their own files by theme
- Require those themes inside the template

NOTE: I chose not to create classes for each theme. When I do so, the Thor commands that are so useful in a Rails template no longer work. For example:

```
run 'bundle'
```

becomes 

```
system "bundle"
```

and really useful commands like 

```
gsub_file 
```

are no longer useable.

### CODE REVIEW

File: lib/database_setup.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  run "cp config/database.yml config/database.example.yml"
+  run "cp config/secrets.yml config/secrets.example.yml"
+  file '.travis.yml', render_file("#{$path}/files/.travis.yml")
+  gsub_file '.travis.yml', /app_name/, app_name
+  file '.rubocop.yml', render_file("#{$path}/files/.rubocop.yml")
+end
+
+def create_database
+  run 'rake db:create'
+end
```
Comment (jahammo2) [inline lib/database_setup.rb:19]: Are all these commands (`run`, `remove_file`) just built in Rails commands? Or are they Ruby? This is dope

**Comment by jahammo2** [comment]
What are `Thor` commands?

File: lib/database_setup.rb (hunk 1 above)
Comment (jahammo2) [inline lib/database_setup.rb:19]: Also, why did you have to put this into its own file? I read the description but I'm still not really sure what you're trying to accomplish

Comment (jahammo2) [inline lib/database_setup.rb:19]: Omg, ignore the last one. I forgot what this project was

**Comment by jahammo2** [comment]
I really like how you are handling the responsibilities. :shipit:

Also...

This is so friggin cool, Ruti. I'm amped about using this in the future.

File: lib/database_setup.rb (hunk 1 above)
Comment (Ru-T) [inline lib/database_setup.rb:19]: @jahammo2 Those commands are Thor -  Rails templates are built on Thor, which describes itself as “a toolkit for building powerful command-line interfaces. It is used in Bundler, Vagrant, Rails and others.” Thor also provides useful documentation on commands available within a Rails template.

**Comment by Ru-T** [comment]
Thanks @jahammo2!!!

**Pull request merged**